### PR TITLE
Rewrite test_exceptions.py to be pytest-style

### PR DESCRIPTION
**Pull request by alexwlchan**
Related to #1160.

### CODE REVIEW

File: test/test_exceptions.py [hunk 1]
```diff
@@ -1,53 +1,37 @@
...
+        HTTPError('foo', IOError('foo')),
+        MaxRetryError(HTTPConnectionPool('localhost'), '/', None),
+        LocationParseError('fake location'),
+        ClosedPoolError(HTTPConnectionPool('localhost'), None),
+        EmptyPoolError(HTTPConnectionPool('localhost'), None),
+        HostChangedError(HTTPConnectionPool('localhost'), '/', None),
+        ReadTimeoutError(HTTPConnectionPool('localhost'), '/', None),
+    ])
+    def test_exceptions(self, exception):
+        assert pickle.loads(pickle.dumps(exception))
```
Comment (Lukasa) [inline test/test_exceptions.py]: Looking at this, one wonders whether there was supposed to be a more concrete assertion here other than just "does not return `None`".

Comment (alexwlchan) [inline test/test_exceptions.py]: Perhaps? I tried changing it to

```python
assert pickle.loads(pickle.dumps(exception)) == exception
```

but that turns out to be False. Maybe that’s indicative of a bug?

Comment (Lukasa) [inline test/test_exceptions.py]: /me shrugs.

Ultimately I have no idea.

Comment (Lukasa) [inline test/test_exceptions.py]: I doubt equality is implemented for these exceptions though, so probably it doesn't matter.

Comment (alexwlchan) [inline test/test_exceptions.py]: Yeah, looks like they don’t have equality helpers. I did make it the test a little stronger anyway, but I don’t think there’s more that can be easily checked beyond that.

**Pull request merged**